src/main/webapp/app/exam/monitoring/exam-action.service.ts

```diff
@@ -78,14 +76,66 @@ export class ExamActionService implements IExamActionService {
      * */
     public unsubscribeForExamAction(exam: Exam): void {
         const topic = EXAM_MONITORING_TOPIC(exam.id!);
-        const userTopic = EXAM_MONITORING_USER_TOPIC(exam.id!);
         this.cachedExamActions.set(exam.id!, []);
         this.initialActionsLoaded.delete(exam.id!);
         this.jhiWebsocketService.unsubscribe(topic);
-        this.jhiWebsocketService.unsubscribe(userTopic);
         this.openExamMonitoringWebsocketSubscriptions.delete(exam.id!);
     }
 
+    /**
+     * Notify all exam monitoring update subscribers with the newest exam monitoring status.
+     * @param exam received or updated exam
+     * @param status whether the monitoring is enabled or not
+     */
+    public notifyExamMonitoringUpdateSubscribers = (exam: Exam, status: boolean) => {
+        const examMonitoringStatusObservable = this.examMonitoringStatusObservables.get(exam.id!);
+        if (!examMonitoringStatusObservable) {
+            this.examMonitoringStatusObservables.set(exam.id!, new BehaviorSubject(status));
+        } else {
+            examMonitoringStatusObservable.next(status);
+        }
+    };
+
+    /**
+     * Checks if a websocket connection for the exam monitoring update to the server already exists.
+     * If not a new one will be opened.
+     *
+     */
+    private openExamMonitoringUpdateWebsocketSubscriptionIfNotExisting(exam: Exam) {
```

Comment:
Other methods in this service only take the examID as a parameter. You might want to unify this

Reply:
I would like to leave it as it is, since this was intended in a way :+1: